src/app/3d/qgs3dmodelsourcelineedit.h

```diff
@@ -0,0 +1,60 @@
+/***************************************************************************
+ qgs3dmodelsourcelineedit.h
+ ---------------------
+ begin                : July 2020
+ copyright            : (C) 2020 by Mathieu Pellerin
+ email                : nirvn dot asia at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGS3DMODELSOURCELINEEDIT_H
+#define QGS3DMODELSOURCELINEEDIT_H
+
+#include "qgsfilecontentsourcelineedit.h"
+#include "qgis_sip.h"
+#include "qgis_3d.h"
+
+#include <QString>
+
+/**
+ * \ingroup 3d
+ * \class Qgs3DModelSourceLineEdit
+ * A line edit widget with toolbutton for setting a 3D model source path.
+ *
+ * Designed for use with QgsSourceCache.
+ *
+ * \since QGIS 3.16
+ */
+class _3D_EXPORT Qgs3DModelSourceLineEdit : public QgsAbstractFileContentSourceLineEdit
```

Comment:
@nirvn this would be my guess why windows build fails... the superfluous _3D_EXPORT may be trying to force DLL import annotation and causing havoc